core/src/main/scala/org/apache/spark/scheduler/Task.scala

```diff
@@ -86,7 +86,14 @@ private[spark] abstract class Task[T](
       (runTask(context), context.collectAccumulators())
     } finally {
       context.markTaskCompleted()
-      TaskContext.unset()
+      try {
+        // Release memory used by this thread for shuffles
```

Comment:
These should both be in try-finally blocks and should log and ignore exceptions to avoid masking task failure exceptions.